MaxText/layers/linears.py

```diff
@@ -521,7 +522,14 @@ def reshape_and_update_weights(self, weights, indices):
   def generate_masks(self, top_k_indices, softmax_probs):
     # calculate expert_capacity = (tokens_per_batch / num_experts) * capacity_factor
     batch_size, seq_len, _ = top_k_indices.shape
-    tokens_per_batch = seq_len * self.num_experts_per_tok
+    cp = self.config.ici_context_parallelism
+    if seq_len % cp != 0:
+      cp = 1
+    sub_seq = seq_len // cp
+
+    top_k_indices = jnp.reshape(top_k_indices, (batch_size, cp, sub_seq, top_k_indices.shape[2]))
```

Comment:
Could we rename `cp` as `context_partitions` or similar? And add a comment about top_k_indices shape, i.e. [batch_size, context_partition, sub_sequence, num_experts_per_tok]?

Reply:
please see the updated comments here.

Reply:
Thanks! Could we add one more line as 

`top_k_indices shape: [batch_size, context_partition, sub_sequence, num_experts_per_tok]` ?